/*
 * Licensed to the Apache Software Foundation (ASF) under one or more
 * contributor license agreements.  See the NOTICE file distributed with
 * this work for additional information regarding copyright ownership.
 * The ASF licenses this file to You under the Apache License, Version 2.0
 * (the "License"); you may not use this file except in compliance with
 * the License.  You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

syntax = "proto3";

option java_multiple_files = true;
package org.apache.airavata.mft.resource.stubs.swift.storage;

message SwiftStorage {
    string storageId = 1;
    string container = 2;
    string name = 3;
    string region = 5;
}

message SwiftStorageListRequest {
    int32 offset = 1;
    int32 limit = 2;
}

message SwiftStorageListResponse {
    repeated SwiftStorage storages = 1;
}

message SwiftStorageGetRequest {
    string storageId = 1;
}

message SwiftStorageCreateRequest {
    string storageId = 1;
    string container = 2;
    string name = 3;
    string region = 5;
}

message SwiftStorageUpdateRequest {
    string storageId = 1;
    string container = 2;
    string name = 3;
    string region = 5;
}

message SwiftStorageUpdateResponse {
    string storageId = 1;
}

message SwiftStorageDeleteRequest {
    string storageId = 1;
}

message SwiftStorageDeleteResponse {
    bool status = 1;
}